packages/arb-avm-cpp/avm/src/machinestate/machinestate.cpp

```diff
@@ -119,12 +116,13 @@ uint256_t MachineState::hash() const {
         oit = to_big_endian(val, oit);
     }
     {
-        if (std::holds_alternative<uint256_t>(staged_message)) {
-            throw std::runtime_error(
-                "Can't get hash of machine with incomplete staged_message");
+        auto message = getStagedMessageTuple();
+        if (!message) {
+            std::cerr << "Can't get hash of machine with incomplete staged_message" << std::endl;
```

Comment:
I think we can drop this `cerr` now that the type system is handling this. Might move it to the cgo layer instead